packages/iTwinUI-react/src/core/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,201 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import '@itwin/itwinui-css/css/dialog.css';
+import DialogTitleBar from './DialogTitleBar';
+import DialogContent from './DialogContent';
+import DialogBackdrop from './DialogBackdrop';
+import { DialogContext, DialogContextProps } from './DialogContext';
+import DialogButtonBar from './DialogButtonBar';
+import { FocusTrap, getDocument, useMergedRefs } from '../utils';
+import { CSSTransition } from 'react-transition-group';
+import cx from 'classnames';
+
+export type DialogProps = {
+  /**
+   * Type of the dialog.
+   * @default 'default'
+   */
+  styleType?: 'default' | 'fullPage';
+  /**
+   * Dialog content.
+   * @example
+   * <Dialog
+   *   isOpen={isOpen}
+   *   onClose={() => setIsOpen(false)}
+   *   backdrop={<Dialog.Backdrop />}
+   *   trapFocus
+   *   preventBodyScroll
+   * >
+   *   <Dialog.TitleBar>My dialog title</Dialog.TitleBar>
+   *   <Dialog.Content>
+   *     Here is my dialog content
+   *   </Dialog.Content>
+   *   <Dialog.ButtonBar>
+   *     <Button styleType='high-visibility'>Confirm</Button>
+   *     <Button>Close</Button>
+   *   </Dialog.ButtonBar>
+   * </Dialog>
+   */
+  children: React.ReactNode;
+  /**
+   * Backdrop element which is used to cover the rest of the page behind the dialog.
+   */
+  backdrop?: React.ReactNode;
+  /**
+   * Traps the focus inside the dialog. This is useful when the dialog is modal.
+   * @default false
+   */
+  trapFocus?: boolean;
+  /**
+   * Prevents body from being scrollable. This is useful when the dialog is modal.
+   * @default false
+   */
+  preventBodyScroll?: boolean;
```

Comment:
Not sure about naming this, whether it is clear. I also thought maybe better rename to `preventDocumentScroll` so that `body` wouldn't mistaken with inner dialog body.

Reply:
Changed